falcon/util/mediatypes.py

```diff
@@ -88,3 +92,221 @@ def parse_header(line: str) -> Tuple[str, Dict[str, str]]:
         return (key.strip(), pdict)
 
     return _parse_header_old_stdlib(line)
+
+
+def _parse_media_type_header(media_type: str) -> Tuple[str, str, dict]:
+    full_type, params = parse_header(media_type)
+
+    # TODO(vytas): Workaround from python-mimeparse by J. Gregorio et al.
+    #   Do we still need this in 2024?
+    # Java URLConnection class sends an Accept header that includes a
+    #   single '*'. Turn it into a legal wildcard.
+    if full_type == '*':
+        full_type = '*/*'
+
+    main_type, separator, subtype = full_type.partition('/')
+    if not separator:
+        raise errors.InvalidMediaType('The media type value must contain type/subtype.')
+
+    return (main_type.strip(), subtype.strip(), params)
+
+
+# TODO(vytas): Should we make these structures public?
+class _MediaType:
```

Comment:
If they are private, does it make any difference?
Are dataclasses faster?

Reply:
they are normal classes, so unlikely. Just more convenient. 

next time I guess

Reply:
I can change to dataclasses, that's not hard. Will just do a quick benchmark if they are not very slow for some reason.

Reply:
Changed to `@dataclasses.dataclass(slots=True)`.

Removed custom `__repr__`, it was only nice for debugging, but just unused code otherwise if/until we publish these classes. Also, `dataclass` gives some `__repr__` for free which is good enough for now.